### PR TITLE
Acorn updates for 1.8.0

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
### Summary

This PR provides updates for the Acorn site config for the 1.8.0 release, which should eventually get merged into develop.

### Testing

Used this config to deploy 1.8.0 on Acorn.

### Applications affected

On Acorn I'm only building a subset of the Unified Env, namely ufs-weather-model-env, global-workflow-env, and a few other things.

### Systems affected

Acorn

### Dependencies

none

### Issue(s) addressed

Part of #1278

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

File: configs/sites/tier1/acorn/packages_intel.yaml [hunk 1]
```diff
@@ -31,7 +31,10 @@
       - '@1.2.1 ~mkl'
     py-numpy:
       require::
-      - '@:1.23 ^openblas'
+      - '^[virtuals=lapack,blas] openblas'
```
Comment (climbfuji) [inline configs/sites/tier1/acorn/packages_intel.yaml:34]: We should probably use this syntax for all sites when we merge this back into develop?

File: configs/sites/tier1/acorn/packages_intel.yaml [hunk 2]
```diff
@@ -31,7 +31,10 @@
...
+      - '@:1.25'
```
Comment (climbfuji) [inline configs/sites/tier1/acorn/packages_intel.yaml:35]: Can you plot with matplotlib and this version with intel@2022?

File: configs/sites/tier1/acorn/compilers.yaml [hunk 3]
```diff
@@ -6,18 +6,17 @@ compilers:
       cxx: CC
       f77: ftn
       fc: ftn
-    flags: {}
+    flags:
+      cflags: -static-libgcc -static-libstdc++ -Bstatic -lstdc++ -Bdynamic -lm -lpthread
```
Comment (climbfuji) [inline configs/sites/tier1/acorn/compilers.yaml:10]: For my curiosity, why is this needed?

File: configs/sites/tier1/acorn/packages_intel.yaml (hunk 2 above)
Comment (DavidHuber-NOAA) [inline configs/sites/tier1/acorn/packages_intel.yaml:35]: Does this `any_of` add to the `@:1.25` or negate it?

File: configs/sites/tier1/acorn/packages_intel.yaml [hunk 4]
```diff
@@ -41,11 +44,16 @@
       require:
       - any_of: ["@2.27.0"]
         when: "%intel@19.1.3.304"
+        message: "2.25.0 is the last version to use C++11 (as opposed to C++17)"
```
Comment (DavidHuber-NOAA) [inline configs/sites/tier1/acorn/packages_intel.yaml]: Is 2.25.0 not required now?  Was the C++ standard updated to 17 with this update?

File: configs/sites/tier1/acorn/packages_intel.yaml (hunk 1 above)
Comment (AlexanderRichert-NOAA) [reply configs/sites/tier1/acorn/packages_intel.yaml:34]: Yeah I think this is the best practices syntax

File: configs/sites/tier1/acorn/packages_intel.yaml (hunk 2 above)
Comment (AlexanderRichert-NOAA) [reply configs/sites/tier1/acorn/packages_intel.yaml:35]: It basically makes it more restrictive for Intel 19 (technically both requirements are being applied).

Comment (climbfuji) [reply configs/sites/tier1/acorn/packages_intel.yaml:35]: But the opposite is needed. We need it restricted to @:1.23 with any newer Intel, and @:1.24 with the older Intel; for GNU, it can be @:1.25.

File: configs/sites/tier1/acorn/compilers.yaml (hunk 3 above)
Comment (AlexanderRichert-NOAA) [reply configs/sites/tier1/acorn/compilers.yaml:10]: There have been issues on Acorn in the past with libstdc++ changing or being inconsistent between login and compute nodes in ways that cause dynamic library loading problems, so basically it's a small risk reducer (and so far everything I've tested compiles fine with it). The 'redundant' -static-libstdc++ and `-Bstatic -lstdc++` are because of some very strange quirk with the Cray wrappers and this was the only way to actually force it to use the static libstdc++ :upside_down_face:

File: configs/sites/tier1/acorn/packages_intel.yaml (hunk 2 above)
Comment (AlexanderRichert-NOAA) [reply configs/sites/tier1/acorn/packages_intel.yaml:35]: It's working with intel@2022 (a.k.a. icc version 2021.6.0) using py-numpy@1.25.2, and intel@19 with py-numpy@1.24.4. So I think I can either leave this line as it is or add a `when: intel@2022`.

Comment (AlexanderRichert-NOAA) [reply configs/sites/tier1/acorn/packages_intel.yaml:35]: Yes

File: configs/sites/tier1/acorn/packages_intel.yaml (hunk 4 above)
Comment (AlexanderRichert-NOAA) [reply configs/sites/tier1/acorn/packages_intel.yaml]: 2.27.0 is no longer in the recipe, so I'm leaving this as 2.25.0 (which is what's now installed for 1.8.0)

**Review by Hang-Lei-NOAA — APPROVED** [review]

File: configs/sites/tier1/acorn/packages_intel.yaml (hunk 2 above)
Comment (climbfuji) [reply configs/sites/tier1/acorn/packages_intel.yaml:35]: Wow. Doesn't work on Hera or S4!

**Review by climbfuji — APPROVED** [review]

**Review by DavidHuber-NOAA — APPROVED** [review]
